x/ecocredit/server/invariants_test.go

```diff
@@ -1,371 +1,268 @@
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+	"github.com/regen-network/regen-ledger/x/ecocredit/mocks"
+	coreserver "github.com/regen-network/regen-ledger/x/ecocredit/server/core"
 )
 
-func setupStore(t *testing.T) (sdk.Context, *sdk.KVStoreKey) {
-	interfaceRegistry := types.NewInterfaceRegistry()
-	ecocredit.RegisterTypes(interfaceRegistry)
-	key := sdk.NewKVStoreKey(ecocredit.ModuleName)
+type baseSuite struct {
+	t            *testing.T
+	db           ormdb.ModuleDB
+	stateStore   api.StateStore
+	ctx          context.Context
+	k            coreserver.Keeper
+	ctrl         *gomock.Controller
+	bankKeeper   *mocks.MockBankKeeper
+	paramsKeeper *mocks.MockParamKeeper
+	storeKey     *sdk.KVStoreKey
+	sdkCtx       sdk.Context
+}
+
+func setupBase(t *testing.T) *baseSuite {
+	// prepare database
+	s := &baseSuite{t: t}
+	var err error
+	s.db, err = ormdb.NewModuleDB(&ecocredit.ModuleSchema, ormdb.ModuleDBOptions{})
+	assert.NilError(t, err)
+	s.stateStore, err = api.NewStateStore(s.db)
+	assert.NilError(t, err)
+
 	db := dbm.NewMemDB()
 	cms := store.NewCommitMultiStore(db)
-	cms.MountStoreWithDB(key, sdk.StoreTypeIAVL, db)
-	err := cms.LoadLatestVersion()
-	require.NoError(t, err)
-	ctx := sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger())
-	return ctx, key
+	s.storeKey = sdk.NewKVStoreKey("test")
+	cms.MountStoreWithDB(s.storeKey, sdk.StoreTypeIAVL, db)
+	assert.NilError(t, cms.LoadLatestVersion())
+	ormCtx := ormtable.WrapContextDefault(ormtest.NewMemoryBackend())
+	s.sdkCtx = sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger()).WithContext(ormCtx)
+	s.ctx = sdk.WrapSDKContext(s.sdkCtx)
+
+	// setup test keeper
+	s.ctrl = gomock.NewController(t)
+	assert.NilError(t, err)
+	s.bankKeeper = mocks.NewMockBankKeeper(s.ctrl)
+	s.paramsKeeper = mocks.NewMockParamKeeper(s.ctrl)
+	s.k = coreserver.NewKeeper(s.stateStore, s.bankKeeper, s.paramsKeeper)
+
+	return s
 }
 
-func TestTradableSupplyInvariants(t *testing.T) {
+func TestBatchSupplyInvariant(t *testing.T) {
 	acc1 := sdk.AccAddress([]byte("account1"))
 	acc2 := sdk.AccAddress([]byte("account2"))
 
 	testCases := []struct {
 		msg           string
-		balances      []*ecocredit.Balance
-		supply        []*ecocredit.Supply
-		basketBalance map[string]math.Dec
+		balances      []*core.BatchBalance
+		supply        []*core.BatchSupply
+		basketBalance map[uint64]math.Dec
 		expBroken     bool
 	}{
 		{
 			"valid test case",
-			[]*ecocredit.Balance{
-				{
-					Address:         acc1.String(),
-					BatchDenom:      "1/2",
-					TradableBalance: "100",
-				},
+			[]*core.BatchBalance{
 				{
-					Address:         acc2.String(),
-					BatchDenom:      "1/2",
-					TradableBalance: "210",
-				},
-				{
-					Address:        acc2.String(),
-					BatchDenom:     "1/2",
-					RetiredBalance: "210",
+					Address:  acc1,
+					BatchId:  1,
+					Tradable: "210",
+					Retired:  "110",
```

Comment:
I think we should include two accounts with a balance for the same batch. It looks like we only test two balances for two separate batches. We should either add here or add a new test case.